src/test/java/org/springframework/samples/petclinic/integration/PetWeightIntegrationTests.java

```diff
@@ -0,0 +1,435 @@
+        ResponseEntity<Void> updateResponse = restTemplate.exchange(
+            baseUrl + "/pets/" + petId,
+            HttpMethod.PUT,
+            updateRequest,
+            Void.class
+        );
+
+        assertEquals(HttpStatus.NO_CONTENT, updateResponse.getStatusCode());
+
+        // 4. Verify weight was updated via REST API
+        ResponseEntity<String> getUpdatedResponse = restTemplate.exchange(
+            baseUrl + "/pets/" + petId,
+            HttpMethod.GET,
+            getRequest,
+            String.class
+        );
+
+        assertEquals(HttpStatus.OK, getUpdatedResponse.getStatusCode());
+        JsonNode updatedPetJson = objectMapper.readTree(getUpdatedResponse.getBody());
+        assertEquals(35.75, updatedPetJson.get("weight").asDouble());
+
+        // 5. Set weight to null via REST API
+        createdPet.setWeight(null);
+        HttpEntity<PetDto> nullWeightRequest = new HttpEntity<>(createdPet, headers);
+
+        ResponseEntity<Void> nullWeightResponse = restTemplate.exchange(
+            baseUrl + "/pets/" + petId,
+            HttpMethod.PUT,
+            nullWeightRequest,
+            Void.class
+        );
+
+        assertEquals(HttpStatus.NO_CONTENT, nullWeightResponse.getStatusCode());
+
+        // 6. Verify weight is null via REST API
+        ResponseEntity<String> getNullWeightResponse = restTemplate.exchange(
+            baseUrl + "/pets/" + petId,
+            HttpMethod.GET,
+            getRequest,
+            String.class
+        );
+
+        assertEquals(HttpStatus.OK, getNullWeightResponse.getStatusCode());
+        JsonNode nullWeightPetJson = objectMapper.readTree(getNullWeightResponse.getBody());
+        assertTrue(nullWeightPetJson.get("weight").isNull());
+
+        // 7. Verify data persistence through service layer
+        Pet persistedPet = clinicService.findPetById(petId);
+        assertNotNull(persistedPet);
+        assertNull(persistedPet.getWeight());
+        assertEquals("TestPet", persistedPet.getName());
+    }
+
+    @Test
+    void testWeightValidationThroughRestApi() throws Exception {
+        PetDto newPet = createTestPetDto();
+
+        // Test invalid negative weight
+        newPet.setWeight(new BigDecimal("-3.00"));
+        HttpEntity<PetDto> invalidRequest = new HttpEntity<>(newPet, headers);
+
+        ResponseEntity<String> invalidResponse = restTemplate.exchange(
+            baseUrl + "/owners/1/pets",
+            HttpMethod.POST,
+            invalidRequest,
+            String.class
+        );
+
+        assertEquals(HttpStatus.BAD_REQUEST, invalidResponse.getStatusCode());
+
+        // Test weight too large
+        newPet.setWeight(new BigDecimal("1000.00"));
+        HttpEntity<PetDto> largeRequest = new HttpEntity<>(newPet, headers);
+
+        ResponseEntity<String> largeResponse = restTemplate.exchange(
+            baseUrl + "/owners/1/pets",
+            HttpMethod.POST,
+            largeRequest,
+            String.class
+        );
```

Comment:
It's best to split tests like this in to several separate test methods.
Like this, if negative weight validation fails – then this too large validation is not even started.

This is exactly what test methods are for – so that you split your testing into separate independent tasks. 

Same with updating pet wieght to a number and updating it to null – can be separate tests.